app/controllers/content_items_controller.rb

```diff
@@ -115,7 +115,9 @@ def set_access_control_allow_origin_header
 
   def set_use_recommended_related_links_header
     response.headers['Vary'] = [response.headers['Vary'], FeatureFlagNames.recommended_related_links].compact.join(', ')
-    response.headers[FeatureFlagNames.recommended_related_links] = Services.feature_toggler.feature_flags.get_feature_flag(FeatureFlagNames.recommended_related_links)
+
+    related_links_response_header = RequestHelper.get_header(FeatureFlagNames.recommended_related_links, request.headers) == Services.feature_toggler.feature_flags.get_feature_flag(FeatureFlagNames.recommended_related_links)
```

Comment:
A couple of variable assignments might help readability here.

e.g. but with proper variable names :)

```ruby
x = FeatureFlagNames.recommended_related_links
y = RequestHelper.get_header(x, request.headers) 
z = Services.feature_toggler.feature_flags.get_feature_flag(x)
response.headers[x] = y == z
```

Reply:
Sure thing